io-sim/test/Test/Control/Monad/Common.hs

```diff
@@ -0,0 +1,509 @@
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Control.Monad.Common where
```

Comment:
The name is misleading as it actually contains properties & utils.  It would be better to have a better distinction between these two.